--- mysql++-1.7.9/examples/resetdb.cc.orig	2004-04-02 00:15:14.000000000 +0200
+++ mysql++-1.7.9/examples/resetdb.cc	2004-04-02 00:21:23.357794224 +0200
@@ -2,6 +2,8 @@
 #include <iostream>
 #include <sqlplus.hh>
 
+using namespace std;
+
 int main (int argc, char *argv[]) {
   Connection connection(use_exceptions);
   try { // the entire main block is one big try block;
--- mysql++-1.7.9/examples/util.cc.orig	2004-04-02 00:15:14.000000000 +0200
+++ mysql++-1.7.9/examples/util.cc	2004-04-02 00:23:34.985783728 +0200
@@ -2,6 +2,8 @@
 #include <iomanip>
 #include "util.hh"
 
+using namespace std;
+
 void print_stock_table(Query& query) {
   query.reset();
   (std::ostream&)query << "select * from stock";
--- mysql++-1.7.9/examples/simple1.cc.orig	2004-04-02 00:15:14.000000000 +0200
+++ mysql++-1.7.9/examples/simple1.cc	2004-04-02 00:25:11.471115728 +0200
@@ -2,6 +2,8 @@
 #include <iomanip>
 #include <sqlplus.hh>
 
+using namespace std;
+
 int main() {
   // The full format for the Connection constructor is
   // Connection(cchar *db, cchar *host="", 
--- mysql++-1.7.9/examples/complic1.cc.orig	2004-04-02 00:15:14.000000000 +0200
+++ mysql++-1.7.9/examples/complic1.cc	2004-04-02 00:29:58.901419680 +0200
@@ -3,6 +3,8 @@
 #include <iomanip>
 #include <sqlplus.hh>
 
+using namespace std;
+
 int main() {
   try { // its in one big try block
 
--- mysql++-1.7.9/examples/fieldinf1.cc.orig	2004-04-02 00:15:14.000000000 +0200
+++ mysql++-1.7.9/examples/fieldinf1.cc	2004-04-02 00:33:04.720170936 +0200
@@ -2,6 +2,7 @@
 #include <iostream>
 #include <iomanip>
 #include <sqlplus.hh>
+using namespace std;
 using std::setw;
 using std::string;
 int main() {
--- mysql++-1.7.9/examples/sinisa_ex.cc.orig	2004-04-02 00:15:14.000000000 +0200
+++ mysql++-1.7.9/examples/sinisa_ex.cc	2004-04-02 00:33:55.489452832 +0200
@@ -6,6 +6,7 @@
 #include <iomanip>
 using std::setw;
 #include <sqlplus.hh>
+using namespace std;
 
 int main() {
   Connection con(use_exceptions);
--- mysql++-1.7.9/examples/updel_x_.cc.orig	2004-04-02 00:15:14.000000000 +0200
+++ mysql++-1.7.9/examples/updel_x_.cc	2004-04-02 00:35:18.111892312 +0200
@@ -1,6 +1,8 @@
 #include <sqlplus.hh>
 #include <string>
+#include <strstream>
 using std::string;
+using namespace std;
 #define MY_DATABASE	"telcent"
 #define MY_TABLE		"nazivi"
 #define MY_HOST    "localhost"
--- mysql++-1.7.9/examples/load_file.cc.orig	2004-04-02 00:15:14.000000000 +0200
+++ mysql++-1.7.9/examples/load_file.cc	2004-04-02 00:36:19.542553432 +0200
@@ -1,7 +1,10 @@
 #include <sys/stat.h>
 #include <fstream>
 #include <mysql++>
-extern int errno;
+#include <errno.h>
+#include <strstream>
+using namespace std;
+
 const char  MY_DATABASE[]="telcent";
 const char  MY_TABLE[]="fax";
 const char  MY_HOST[]="localhost";
--- mysql++-1.7.9/examples/cgi_image.cc.orig	2004-04-02 00:15:14.000000000 +0200
+++ mysql++-1.7.9/examples/cgi_image.cc	2004-04-02 00:40:36.721456296 +0200
@@ -1,4 +1,5 @@
 #include <sqlplus.hh>
+using namespace std;
 
 #define MY_DATABASE	"telcent"
 #define MY_TABLE		"fax"
