--- mysql++-1.7.9/configure.in.orig	2004-04-02 00:46:34.000000000 +0200
+++ mysql++-1.7.9/configure.in	2004-04-02 01:18:56.645814648 +0200
@@ -37,7 +37,7 @@
 
 AC_ARG_WITH(mysql-lib,
     [  --with-mysql-lib=<path> directory path of MySQL library installation],
-    [MYSQL_lib_check=$with_mysql/lib/mysql])
+    [MYSQL_lib_check=$with_mysql])
 
 AC_ARG_WITH(mysql-include,
     [  --with-mysql-include=<path>
